back-end/placementdata/views.py

```diff
@@ -0,0 +1,55 @@
+# from django.shortcuts import render,get_object_or_404
+from rest_framework.response import Response
+from rest_framework import status 
+from .models import PlacementData
+from .serializers import PlacementDataSerializer
+from .services import *
+from rest_framework.views import APIView
+
+
+class placements_show(APIView):
+    def get(self,request,*args,**kwargs):
+        serialized_data = get_placement_data()
+        return Response(serialized_data.data)
+
+class placement_create(APIView):
+    def post(self,request,*args,**kwargs):
+        serializedData = PlacementDataSerializer(data = request.data,context={'user_id': request.user})
+        if serializedData.is_valid():
+            serializedData.save()
+            return Response(serializedData.data)
+        else:
+            return Response(serializedData.errors)
+
+class placement_view(APIView):
+    def get(self,request,pk,*args,**kwargs):
+        try:
+            current_resource = get_placement_data_by_id(pk)
+        except:
+            return Response({'errors':'Invalid Referennce'},status = status.HTTP_404_NOT_FOUND)
+        serializer = PlacementDataSerializer(current_resource)
+        return Response(serializer.data)
+    
+    def put(self,request,pk,*args,**kwargs):
+        try:
+            current_resource = get_placement_data_by_id(pk)
+        except:
+            return Response({'errors':'Invalid Referennce'},status = status.HTTP_404_NOT_FOUND)    
```

Comment:
Make errors properly formatted, Will suggest making an Error class, which contains the error code, and error messages in an array. Then every time you wish to send an error in Response, create an object of that class with an error code and response. Error codes you can assign starting from 100, like 100, 101, 102 and so on. This will help to debug errors later in the development cycle easily.